play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/accordion/Accordion.scala

```diff
@@ -39,18 +39,25 @@ object Accordion {
 
   def defaultObject: Accordion = Accordion()
 
+  // The purpose of this custom reader is to serve as a fallback for mapping JSON fields that have different names than the corresponding fields in the case class.
+  def customToggleOptionReads(fieldName: String, fallbackFieldName: String): Reads[Option[String]] =
+    (__ \ fieldName).readNullable[String].flatMap {
+      case Some(text) if text != null => Reads.pure(Some(text))
+      case _                          => (__ \ fallbackFieldName).readNullable[String]
+    }
+
   implicit def jsonReads: Reads[Accordion] = (
     (__ \ "id").readWithDefault[String](defaultObject.id) and
       (__ \ "headingLevel").readWithDefault[Int](defaultObject.headingLevel) and
       (__ \ "classes").readWithDefault[String](defaultObject.classes) and
       (__ \ "attributes").readWithDefault[Map[String, String]](defaultObject.attributes)(attributesReads) and
       (__ \ "items").readWithDefault[Seq[Section]](defaultObject.items)(forgivingSeqReads[Section]) and
-      (__ \ "hideSectionText").readNullable[String] and
-      (__ \ "hideAllSectionsText").readNullable[String] and
-      (__ \ "hideSectionAriaLabelText").readNullable[String] and
-      (__ \ "showSectionText").readNullable[String] and
-      (__ \ "showAllSectionsText").readNullable[String] and
-      (__ \ "showSectionAriaLabelText").readNullable[String] and
+      customToggleOptionReads("hideSectionText", "hideSection") and
+      customToggleOptionReads("hideAllSectionsText", "hideAllSections") and
+      customToggleOptionReads("hideSectionAriaLabelText", "hideSectionAriaLabel") and
+      customToggleOptionReads("showSectionText", "showSection") and
+      customToggleOptionReads("showAllSectionsText", "showAllSections") and
+      customToggleOptionReads("showSectionAriaLabelText", "showSectionAriaLabel") and
```

Comment:
I like this idea, but I'm concerned it could evolve into a separate side project and may not be as straightforward as it appears. Perhaps we should add it to the tech debt board and revisit it later if we all agree it's something we want to address.